server/room-battle.ts

```diff
@@ -1119,29 +1118,40 @@ export class RoomBattle extends RoomGames.RoomGame {
 
 		// @ts-ignore
 		this.room = null;
+		if (this.dataResolvers) {
+			for (const [, reject] of this.dataResolvers) {
+				// reject the promise, make whatever function called it return undefined
+				reject('Battle was destroyed.');
+			}
+		}
 	}
 	async getTeam(user: User) {
 		const id = user.id;
 		const player = this.playerTable[id];
 		if (!player) return;
 		void this.stream.write(`>requestteam ${player.slot}`);
-		const teamDataPromise = new Promise<string[]>(resolve => {
+		const teamDataPromise = new Promise<string[]>((resolve, reject) => {
 			if (!this.dataResolvers) this.dataResolvers = [];
-			this.dataResolvers.push(resolve);
+			this.dataResolvers.push([resolve, reject]);
+		}).catch(e => {
+			if (!e.message.includes("Battle was destroyed.")) throw e;
```

Comment:
```suggestion
```

I'm surprised this doesn't crash, considering you tried to reject with a string.

I suspect you might not have tested it.

Reply:
But anyway you shouldn't catch errors here at all, I think.